kubeedge-counter-demo/README.md

```diff
@@ -0,0 +1,81 @@
+# KubeEdge Counter Demo
+
+## Description
+
+Counter is a pseudo device that user can run this demo without any extra physical devices.
+
+Counter run at edge side, and user can control it in web from cloud side, also can get counter value in web from cloud side.
+
+![work flow](./images/work-flow.png)
+
+
+## Prerequisites
+
+### Hardware Prerequisites
+
+* RaspBerry PI (RaspBerry PI 4 has been used for this demo).
+
+### Software Prerequisites
+
+* A running Kubernetes cluster.
+
+  *NOTE*:
+
+  add follows `--insecure-port=8080` and `--insecure-bind-address=0.0.0.0` options into */etc/kubernetes/manifests/kube-apiserver.yaml*
+
+* KubeEdge [1.2.1](https://github.com/kubeedge/kubeedge/tree/v1.2.1)  
+  See [instructions](http://docs.kubeedge.io/en/latest/index.html) on how to setup KubeEdge.
+
+* MQTT Broker is running on Raspi.
+
+## Steps to run the demo
+
+### Create the device model and device instance for the counter
+
+With the Device CRD APIs now installed in the cluster, we create the device model and instance for the counter using the yaml files.
+
+```console
+ cd $GOPATH/src/github.com/kubeedge/examples/kubeedge-counter-demo/crds
+ kubectl create -f kubeedge-counter-model.yaml
+ kubectl create -f kubeedge-counter-instance.yaml
+```
+
+*NOTE*:
+in *crds/kubeedge-counter-instance.yaml*, make sure *nodeSelector* value match with correct edge node name.
+
+### Run KubeEdge Web App
+
+The KubeEdge Web App runs in a VM on cloud.
+
+```console
+ cd $GOPATH/src/github.com/kubeedge/examples/kubeedge-counter-demo/crds
+ kubectl create -f kubeedge-web-controller-app.yaml
+```
+
+### Run KubeEdge Pi Counter App
+
+The KubeEdge Counter App run in raspi.
+
+```console
+ cd $GOPATH/src/github.com/kubeedge/examples/kubeedge-counter-demo/crds
+ kubectl create -f kubeedge-pi-counter-app.yaml
```

Comment:
Actually this is what we already discussed offline, but i like to hear from the community if either deployment or binary install is better for kubeedge example. Since this is kubeedge, edge device could be resource constrained platform which cannot support container or something. if that is the major focus, this should be binary deployment?

Reply:
If it's binary deployment, how to manage the binary/process at edge? Or has any idea for managing binary deployment process? Kubedge is currently focused on managing edge containers and we can use the lightweight runtime like cri-o and runc. @fujitatomoya

Reply:
actually that is what i thought, so i am good to go with the current container deployment.
with KubeEdge, it does not actually require user to run container on edge, as long as edgecore is running, that is good enough. I was thinking that this is one of the use cases, and wondering if  this example should emphasize this case.